cmd/sso.go

```diff
@@ -0,0 +1,202 @@
+package cmd
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+
+	"github.com/supabase/cli/internal/sso/create"
+	"github.com/supabase/cli/internal/sso/get"
+	"github.com/supabase/cli/internal/sso/info"
+	"github.com/supabase/cli/internal/sso/list"
+	"github.com/supabase/cli/internal/sso/remove"
+	"github.com/supabase/cli/internal/sso/update"
+	"github.com/supabase/cli/internal/utils"
+)
+
+var (
+	ssoCmd = &cobra.Command{
+		GroupID: groupManagementAPI,
+		Use:     "sso",
+		Short:   "Manage Single Sign-On (SSO) authentication for projects",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.Root().PersistentPreRunE(cmd, args); err != nil {
+				return err
+			}
+			ctx, _ := signal.NotifyContext(cmd.Context(), os.Interrupt)
+			cmd.SetContext(ctx)
+
+			fsys := afero.NewOsFs()
+			if err := PromptProjectRef(fsys, cmd); err != nil {
+				return err
+			}
+
+			// Validate project ref
+			if !utils.ProjectRefPattern.MatchString(projectRef) {
+				return errors.New("Invalid project ref format. Must be like `abcdefghijklmnopqrst`.")
+			}
+			return nil
+		},
+	}
+
+	ssoMetadataFile         string
+	ssoMetadataURL          string
+	ssoSkipURLValidation    bool
+	ssoMetadata             bool
+	ssoAttributeMappingFile string
+	ssoDomains              []string
+	ssoAddDomains           []string
+	ssoRemoveDomains        []string
+	ssoOutput               = utils.EnumFlag{
+		Allowed: utils.OutputDefaultAllowed,
+		Value:   utils.OutputPretty,
+	}
+
+	ssoAddCmd = &cobra.Command{
+		Use:     "add <type = saml> [flags]",
```

Comment:
Might be a bit late to ask but can we make this an enum flag? ie. `sso add --type saml`

```go
	ssoType = utils.EnumFlag{
		Allowed: []string{"saml"},
		Value:   "saml",
	}
```

This would give us automatic validation and default value support.

Also, the `[flags]` part of UseLine will be added automatically by cobra

```suggestion
		Use:     "add",
```

Reply:
Sure no problem. I was a bit skeptical of the positional argument myself.